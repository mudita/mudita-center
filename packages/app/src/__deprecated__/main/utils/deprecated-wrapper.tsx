/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
