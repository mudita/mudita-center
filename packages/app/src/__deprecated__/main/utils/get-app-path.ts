/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import path from "path"
import packageInfo from "../../../../package.json"

export enum AppType {
  Main = "main",
  Renderer = "renderer",
}
export type AppResolver = () => {
  type: AppType
  app: { getPath: (arg0: string) => string }
}

export const resolve: AppResolver = () => {
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const isRenderer = require("is-electron-renderer")
  // FIXME: Using remote module in renderer process isn't a good pattern.
  //  You can read more in https://github.com/electron/remote#migrating-from-remote
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const app = isRenderer
    ? // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("@electron/remote").app
    : // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("electron").app
  const type = isRenderer ? AppType.Renderer : AppType.Main
  return {
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    app,
    type,
  }
}

const getAppPath = (): string => {
  try {
    return path.join(resolve().app.getPath("appData"), packageInfo.name)
  } catch {
    return ""
  }
}

export default getAppPath
