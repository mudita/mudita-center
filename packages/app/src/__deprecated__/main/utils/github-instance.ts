/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import axios from "axios"

// It's required only for development when API rate limits may exceed
// https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting
const githubToken = process.env.OS_UPDATE_SERVER_ACCESS_TOKEN

export const githubInstance = axios.create({
  headers: {
    ...(githubToken ? { Authorization: `token ${githubToken}` } : {}),
  },
})
