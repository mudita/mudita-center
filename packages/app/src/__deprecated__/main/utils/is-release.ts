/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import isVersionMatch from "App/overview/helpers/is-version-match"

export const isRelease = (version: string): boolean => {
  if(isVersionMatch(version)){
    return /(\d+\.){2}\d+$/i.test(version)
  } else {
    return false
  }
}
