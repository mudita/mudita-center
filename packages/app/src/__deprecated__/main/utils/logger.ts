/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { createLogger, format, Logger, transports } from "winston"
import path from "path"
import getAppPath, {
  AppResolver,
  AppType,
  resolve,
} from "App/__deprecated__/main/utils/get-app-path"
import { getMetadataStore } from "App/metadata/containers"

const DailyRotateFile = require("winston-daily-rotate-file")
const RollbarTransport = require("winston-transport-rollbar-3")
const testing = process.env.NODE_ENV === "test"
const { combine, timestamp, printf, colorize, simple } = format

// TODO: This type should only have some of the log methods instead of using the full Winstonâ€™s interface.
export type AppLogger = Logger & {
  enableRollbar: () => void
  disableRollbar: () => void
  updateMetadata: () => void
}

export const logsPath = path.join(getAppPath(), "logs")

let defaultMeta: any | null = {}

const createDailyRotateFileTransport = (
  type: AppType,
  defaultMetadata: any
) => {
  const format = combine(
    timestamp(),
    printf(({ level, message, timestamp }) => {
      const paddedProcess = `[${type}]`.padEnd(10, " ")
      const paddedLevel = `[${level}]:`.padEnd(8, " ")
      const metadata = `\n[metadata]:\n${Object.entries(defaultMetadata)
        .map(([key, value]) => (value ? `${key}: ${value}` : ""))
        .filter((item) => item)
        .join("\n")}`
      return `${paddedProcess} [${timestamp}] ${paddedLevel} ${message} ${metadata}`
    })
  )
  return new DailyRotateFile({
    dirname: logsPath,
    filename: "mc-%DATE%",
    extension: ".log",
    datePattern: "YYYY-MM-DD",
    maxFiles: 10,
    maxSize: "1d",
    level: "info",
    format,
  })
}

const consoleTransport = new transports.Console({
  level: testing ? "emerg" : "silly",
  silent: testing,
  format: combine(colorize(), simple()),
})

let dailyRotateFileTransport = createDailyRotateFileTransport(
  AppType.Main,
  defaultMeta
)

// TODO: test this. https://appnroll.atlassian.net/browse/PDA-764
export const createAppLogger = (resolveApp: AppResolver): AppLogger => {
  const { app, type } = resolveApp()
  dailyRotateFileTransport = createDailyRotateFileTransport(
    type,
    defaultMeta
  )

  const rollbarTransport = new RollbarTransport({
    rollbarConfig: {
      accessToken: process.env.ROLLBAR_TOKEN || "test",
      captureUncaught: true,
      captureUnhandledRejections: true,
      enabled: false,
      payload: {
        environment: process.env.NODE_ENV,
      },
      scrubTelemetryInputs: true,
      autoInstrument: {
        network: false,
        log: false,
        dom: true,
        navigation: true,
        connectivity: false,
        contentSecurityPolicy: false,
        errorOnContentSecurityPolicy: false,
        scrubFields: ["phoneNumber"],
      },
      captureIp: "anonymize",
    },
    level: "warn",
  })

  const appLogger = createLogger({
    level: "info",
    format: format.combine(format.metadata(), format.json()),
    defaultMeta,
    exitOnError: false,
    transports: [...(app ? [dailyRotateFileTransport] : []), consoleTransport],
  })
  const enableRollbar = () => {
    appLogger.add(rollbarTransport)
    rollbarTransport.rollbar.configure({ enabled: true })
  }
  const disableRollbar = () => {
    appLogger.remove(rollbarTransport)
    rollbarTransport.rollbar.configure({ enabled: false })
  }
  const updateMetadata = () => {
    const tmpMetadata = getMetadataStore().metadata ?? {}
    defaultMeta = Object.fromEntries(tmpMetadata)
    appLogger.defaultMeta = defaultMeta
    appLogger.remove(dailyRotateFileTransport)
    dailyRotateFileTransport = createDailyRotateFileTransport(type, defaultMeta)
    appLogger.add(dailyRotateFileTransport)
  }
  return Object.assign(appLogger, {
    enableRollbar,
    disableRollbar,
    updateMetadata
  })
}
const logger = createAppLogger(resolve)
export default logger
