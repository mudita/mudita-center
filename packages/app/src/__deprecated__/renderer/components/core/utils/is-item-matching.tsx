/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export type IsItemMatching = (item: any, searchString: string) => boolean
export const isItemValueMatching: IsItemMatching = (
  itemValue: string,
  search
) => itemValue.toLowerCase().includes(search.toLowerCase())
