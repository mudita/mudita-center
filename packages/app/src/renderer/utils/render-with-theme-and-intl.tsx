/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
