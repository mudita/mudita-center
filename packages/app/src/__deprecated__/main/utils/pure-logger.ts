/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { ConsoleLogger } from "@mudita/pure"
import logger from "App/__deprecated__/main/utils/logger"
import { flags, Feature } from "App/feature-flags"

export interface ScrubProps {
  body?: any
  endpoint: number
  status: number
  uuid: number
}

class PureLogger implements ConsoleLogger {
  info(message: string): void {
    if (flags.get(Feature.LogsScrubbingEnabled)) {
      logger.info(this.scrub(message))
    } else {
      logger.info(this.scrubFileSystemData(message))
    }
  }

  private scrub(message: string): string {
    try {
      const messageParsed = JSON.parse(message) as ScrubProps[]
      if (messageParsed[0].body) {
        return JSON.stringify(
          [{ ...messageParsed[0], body: "scrubbed" }],
          null,
          2
        )
      } else {
        return message
      }
    } catch {
      return message
    }
  }

  private scrubFileSystemData(message: string): string {
    try {
      const messageParsed = JSON.parse(message) as ScrubProps[]
      if (
        messageParsed[0].body &&
        messageParsed[0].endpoint === 3 &&
        Boolean(messageParsed[0].body?.data)
      ) {
        return JSON.stringify(
          [
            {
              ...messageParsed[0],
              body: { ...messageParsed[0].body, data: "scrubbed" },
            },
          ],
          null,
          2
        )
      } else {
        return message
      }
    } catch {
      return message
    }
  }
}

export default PureLogger
