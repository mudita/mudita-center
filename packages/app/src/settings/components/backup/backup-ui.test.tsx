/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { renderWithThemeAndIntl } from "App/__deprecated__/renderer/utils/render-with-theme-and-intl"
import React from "react"
import BackupUI from "App/settings/components/backup/backup-ui.component"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <BackupUI backupLocation="C:/MuditaOs" />
  )
  expect(container).toMatchSnapshot()
})

test("passed prop is displayed in correct place", () => {
  const location = "C:/MuditaOs"
  const { getByTestId } = renderWithThemeAndIntl(
    <BackupUI backupLocation={location} />
  )
  expect(getByTestId("backup-location")).toHaveTextContent(location)
})
