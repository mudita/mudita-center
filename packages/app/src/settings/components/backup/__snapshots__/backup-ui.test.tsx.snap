// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  color: #000000;
}

.c6 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.57;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  color: #6a6a6a;
}

.c12 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.67;
  font-weight: 400;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-transform: uppercase;
  color: #000000;
}

.c4 {
  grid-area: Name;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c7 {
  grid-area: Message;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c8 {
  grid-area: Actions;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  padding: 0.1rem 0.7rem;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  overflow: hidden;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  width: 17.6rem;
  height: 4rem;
  border-radius: 0.4rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c10:hover {
  background: #000000;
}

.c10 p {
  color: #ffffff;
}

.c10 g path {
  fill: #ffffff;
}

.c13 {
  white-space: nowrap;
}

.c0 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 16.4rem;
  border-bottom: solid 0.1rem #d2d6db;
  height: 7.2rem;
  max-height: 7.2rem;
}

.c2 {
  grid-area: Checkbox;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  margin-left: 3.2rem;
  margin-bottom: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 20rem;
  border-bottom: solid 0.1rem #d2d6db;
}

.c9 {
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
}

.c11 {
  padding: 0;
}

<div>
  <section
    class=""
  >
    <div
      class="c0 c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3 c4 c5"
          color="primary"
        >
          [value] module.settings.backupLabel
          <p
            class="c6 c7"
            color="secondary"
            data-testid="backup-location"
          >
            C:/MuditaOs
          </p>
        </p>
      </div>
      <div
        class="c8 c9"
      >
        <button
          class="c10 c11"
          size="1"
          type="button"
        >
          <p
            class="c12 c13"
            color="primary"
          >
            [value] module.settings.backupButtonLabel
          </p>
        </button>
      </div>
    </div>
  </section>
</div>
`;
